grakn-graph/src/test/java/ai/grakn/graph/internal/EntityTypeTest.java

```diff
@@ -579,4 +581,38 @@ public void testChangingSuperTypeBackToMetaType(){
         assertEquals(graknGraph.getMetaEntityType(), entityTypeB.superType());
 
     }
+
+    @Test
+    public void checkSubTypeCachingUpdatedCorrectlyWhenChangingSuperTypes(){
+        EntityType e1 = graknGraph.putEntityType("entityType1");
+        EntityType e2 = graknGraph.putEntityType("entityType2").superType(e1);
+        EntityType e3 = graknGraph.putEntityType("entityType3").superType(e1);
+        EntityType e4 = graknGraph.putEntityType("entityType4").superType(e1);
+        EntityType e5 = graknGraph.putEntityType("entityType5");
+        EntityType e6 = graknGraph.putEntityType("entityType6").superType(e5);
+
+        assertEquals(4, e1.subTypes().size());
+        assertTrue("e1 subtypes does not contain e1", e1.subTypes().contains(e1));
+        assertTrue("e1 subtypes does not contain e2", e1.subTypes().contains(e2));
+        assertTrue("e1 subtypes does not contain e3", e1.subTypes().contains(e3));
+        assertTrue("e1 subtypes does not contain e4", e1.subTypes().contains(e4));
```

Comment:
`assertThat(e1.subTypes(), containsInAnyOrder(e1, e2, e3, e4));`
will cover all of these (including the size check).